src/pointer.rs

```diff
@@ -475,14 +475,46 @@ impl Pointer {
     /// ## Examples
     /// ```
     /// let ptr = jsonptr::Pointer::from_static("/foo");
-    /// let barbaz = jsonptr::Pointer::from_static("/bar/baz");
-    /// assert_eq!(ptr.concat(&barbaz), "/foo/bar/baz");
+    /// let other = jsonptr::Pointer::from_static("/bar/baz");
+    /// assert_eq!(ptr.concat(other), "/foo/bar/baz");
     /// ```
     pub fn concat(&self, other: &Pointer) -> PointerBuf {
         let mut buf = self.to_buf();
         buf.append(other);
         buf
     }
+
+    //  Returns the length of `self` in encoded format.
+    ///
+    /// This length is in bytes, not [`char`]s or graphemes. In other words, it might
+    /// not be what a human considers the length of the string.
+    ///
+    /// ## Examples
+    /// ```
+    /// let mut ptr = jsonptr::Pointer::from_static("/foo/bar").to_buf();
+    /// assert_eq!(ptr.len(), 8);
+    ///
+    /// ptr.push_back("~");
+    /// assert_eq!(ptr.len(), 11);
+    ///
+    /// ```
+    pub fn len(&self) -> usize {
```

Comment:
I think this is sane. One can always do `ptr.tokens().count()` for the alternative definition of "length".

I see that you've copied the doc-comment from `str::len`. This is mostly accurate, since we do indeed just delegate to that method, but the semantics is not quite right. I think we could go with something simpler instead, along the lines of:

```
    /// This length expresses the byte count of the underlying string that represents the RFC 6091 Pointer. See also [`std::str::len`].
```

Although the caveat with char/graphemes applies, it's less important for us than the relationship with `std::str`.